packages/tools/bot/bot.ts

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env node
+
+import Utils from '@kaetram/server/ts/util/utils';
+import io from 'socket.io-client';
+import { each, isArray } from 'lodash';
+
+const log = console;
+
+const config = { debugLevel: 'all', gver: 1 };
+
+class Entity {
+    id;
+    x: number;
+    y: number;
+    connection: SocketIOClient.Socket;
+
+    constructor(id, x: number, y: number, connection: SocketIOClient.Socket) {
+        this.id = id;
+        this.x = x;
+        this.y = y;
+
+        this.connection = connection;
+    }
+}
+
+// export default Entity;
+
+class Bot {
+    bots: Entity[];
+    botCount: number;
+
+    constructor() {
+        this.bots = [];
+        this.botCount = 300;
+
+        this.load();
+    }
+
+    load(): void {
+        const connecting = setInterval(() => {
+            this.connect();
+
+            this.botCount--;
+
+            if (this.botCount < 1) clearInterval(connecting);
+        }, 100);
+
+        setInterval(() => {
+            each(this.bots, (bot) => {
+                this.move(bot);
+
+                if (Utils.randomInt(0, 50) === 10) this.talk(bot);
+            });
+        }, 2000);
+    }
+
+    connect(): void {
+        const connection = io('ws://127.0.0.1:9001', {
+            forceNew: true,
+            reconnection: false
+        });
+
+        connection.on('connect', () => {
+            log.info('Connection established...');
+
+            connection.emit('client', {
+                gVer: config.gver,
+                cType: 'HTML5',
+                bot: true
+            });
+        });
+
+        connection.on('connect_error', () => {
+            log.info('Failed to establish connection.');
+        });
+
+        connection.on('message', (message) => {
+            if (message.startsWith('[')) {
+                const data = JSON.parse(message);
+
+                if (data.length > 1)
+                    each(data, (msg) => {
+                        this.handlePackets(connection, msg);
+                    });
+                else this.handlePackets(connection, JSON.parse(message).shift());
+            } else this.handlePackets(connection, message, 'utf8');
+        });
+
+        connection.on('disconnect', () => {
+            //
+        });
+    }
+
+    handlePackets(connection: SocketIOClient.Socket, message: never[], type?: string): void {
```

Comment:
Why `message` is an array of `never`? It means that the array will always be empty so there is no need for this argument.